common/djangoapps/student/tests/test_admin_views.py

```diff
@@ -324,15 +325,23 @@ def setUp(self):
         """Setup."""
         super(LoginFailuresAdminTest, self).setUp()
         self.client.login(username=self.user.username, password='test')
-        user = UserFactory.create()
+        self.user2 = UserFactory.create(username=u'Zażółć gęślą jaźń')
         LoginFailures.objects.create(user=self.user, failure_count=10, lockout_until=datetime.datetime.now())
-        LoginFailures.objects.create(user=user, failure_count=2)
+        LoginFailures.objects.create(user=self.user2, failure_count=2)
 
     def tearDown(self):
         """Tear Down."""
         super(LoginFailuresAdminTest, self).tearDown()
         LoginFailures.objects.all().delete()
 
+    def test_unicode_username(self):
+        """
+        Test if `__str__` method behaves correctly for unicode username.
+        It shouldn't raise `TypeError`.
+        """
+        str(LoginFailures.objects.get(user=self.user))
+        str(LoginFailures.objects.get(user=self.user2))
```

Comment:
Can we check that str is returning the result we expect?

Reply:
Sure, done :+1: